src/System.Diagnostics.Process/tests/ProcessTests.Unix.cs

```diff
@@ -304,9 +304,93 @@ public void TestStartOnOSXWithBadFormat()
             }
         }
 
+        [Fact]
+        public void TestStartWithNonExistingUserThrows()
+        {
+            Process p = CreateProcessPortable(RemotelyInvokable.Dummy);
+            p.StartInfo.UserName = "DoesNotExist";
+            Assert.Throws<Win32Exception>(() => p.Start());
+        }
+
+        /// <summary>
+        /// Tests when running as a normal user and starting a new process as the same user
+        /// works as expected.
+        /// </summary>
+        [Fact]
+        public void TestStartWithNormalUser()
+        {
+            TestStartWithUserName();
+        }
+
+        /// <summary>
+        /// Tests when running as root and starting a new process as a normal user,
+        /// the new process doesn't have elevated privileges.
+        /// </summary>
```

Comment:
Should we test that the child process when started as normal user, cannot escalate to root again? This is possible if the implementation (accidentally) used setegid() and seteuid() instead of setgid() and setuid().

Reply:
I've added this test.